File: Reporting/VB/ReportMainDemo/BaseForms/RibbonReportDesigner/MainForm.vb
Imports DevExpress.XtraReports.UserDesigner

Namespace RibbonReportDesigner

    Public Partial Class MainForm
        Inherits MainFormBase

        Public Sub New()
            InitializeComponent()
        End Sub

        Public Overrides ReadOnly Property ActiveXRDesignPanel As XRDesignPanel
            Get
                Return reportDesigner1.ActiveDesignPanel
            End Get
        End Property

        Public Overrides Sub OpenReport(ByVal newReport As DevExpress.XtraReports.UI.XtraReport)
            reportDesigner1.OpenReport(newReport)
        End Sub

        Public Sub CreateNewReport()
            reportDesigner1.CreateNewReport()
        End Sub
    End Class
End Namespace

File: Reporting/CS/ReportMainDemo/BaseForms/RibbonReportDesigner/MainForm.cs
using DevExpress.XtraReports.UserDesigner;

namespace RibbonReportDesigner {
    public partial class MainForm : MainFormBase {
        public MainForm() {
            InitializeComponent();
        }
        public override XRDesignPanel ActiveXRDesignPanel {
            get { return reportDesigner1.ActiveDesignPanel; }
        }
        public override void OpenReport(DevExpress.XtraReports.UI.XtraReport newReport) {
            reportDesigner1.OpenReport(newReport);
        }
        public void CreateNewReport() {
            reportDesigner1.CreateNewReport();
        }
    }
}
